src/openrct2/scripting/ScriptEngine.cpp

```diff
@@ -771,6 +777,44 @@ DukValue ScriptEngine::PositionToDuk(const CoordsXYZ& position)
     return DukValue::take_from_stack(ctx);
 }
 
+constexpr static const char* ExpenditureTypes[] = { "ride_construction",
+                                                    "ride_runningcosts",
+                                                    "land_purchase",
+                                                    "landscaping",
+                                                    "park_entrance_tickets",
+                                                    "park_ride_tickets",
+                                                    "shop_sales",
+                                                    "shop_stock",
+                                                    "food_drink_sales",
+                                                    "food_drink_stock",
+                                                    "wages",
+                                                    "marketing",
+                                                    "research",
+                                                    "interest" };
+
+std::string_view ScriptEngine::ExpenditureTypeToString(ExpenditureType expenditureType)
+{
+    auto index = static_cast<size_t>(expenditureType);
+    if (index < std::size(ExpenditureTypes))
+    {
+        return ExpenditureTypes[index];
+    }
+    return {};
+}
+
+ExpenditureType ScriptEngine::StringToExpenditureType(const std::string_view expenditureType)
+{
+    size_t i = 0;
+    for (auto sz : ExpenditureTypes)
+    {
+        if (expenditureType == sz)
+        {
+            return static_cast<ExpenditureType>(i);
```

Comment:
You never increase `i`, so this will always return 0 if the expenditure type exists.

Reply:
Well spotted! Thanks!